Content.Client/Access/UI/AccessLevelControl.xaml.cs

```diff
@@ -0,0 +1,56 @@
+using System.Linq;
+using Robust.Client.AutoGenerated;
+using Robust.Client.UserInterface;
+using Robust.Client.UserInterface.Controls;
+using Robust.Client.UserInterface.XAML;
+using Robust.Shared.Prototypes;
+using Content.Shared.Access;
+using Content.Shared.Access.Systems;
+
+namespace Content.Client.Access.UI;
+
+[GenerateTypedNameReferences]
+public sealed partial class AccessLevelControl : GridContainer
+{
+    public readonly Dictionary<string, Button> ButtonsList = new();
+
+    public AccessLevelControl(List<string> accessLevels, IPrototypeManager prototypeManager)
+    {
+        RobustXamlLoader.Load(this);
+        foreach (var access in accessLevels)
+        {
+            if (!prototypeManager.TryIndex<AccessLevelPrototype>(access, out var accessLevel))
+            {
+                Logger.Error($"Unable to find accesslevel for {access}");
+                continue;
+            }
+
+            var newButton = new Button
+            {
+                Text = GetAccessLevelName(accessLevel),
+                ToggleMode = true,
+            };
+            AddChild(newButton);
+            ButtonsList.Add(accessLevel.ID, newButton);
+        }
```

Comment:
Put this code in a new method so that it can be populated after instantiation. Usually other controls call it `Populate`.

This should allow the AccessLevelControl to be defined inside XAML and the Populate method the be called in the constructor. This makes it so that the AccessLevelControl doesn't need to be created in the constructor.

Reply:
Oh, didn't know that. It's very convenient, thank you!